src/TestSuite/EmailTrait.php

```diff
@@ -0,0 +1,166 @@
+    }
+    /**
+     *
+     * Asserts that no emails were sent
+     *
+     * @param string $message Message
+     * @return void
+     */
+    public function assertNoMailSent($message = null)
+    {
+        $this->assertThat(null, new NoMailSentConstraint(), $message);
+    }
+
+    /**
+     * Asserts an email at a specific index was sent to an address
+     *
+     * @param int $at Email index
+     * @param int $address Email address
+     * @param string $message Message
+     * @return void
+     */
+    public function assertMailSentToAt($at, $address, $message = null)
+    {
+        $this->assertThat($address, new MailSentToConstraint($at), $message);
+    }
+
+    /**
+     * Asserts an email at a specific index was sent from an address
+     *
+     * @param int $at Email index
+     * @param int $address Email address
+     * @param string $message Message
+     * @return void
+     */
+    public function assertMailSentFromAt($at, $address, $message = null)
+    {
+        $this->assertThat($address, new MailSentFromConstraint($at), $message);
+    }
+
+    /**
+     * Asserts an email at a specific index contains expected contents
+     *
+     * @param int $at Email index
+     * @param int $contents Contents
+     * @param string $message Message
+     * @return void
+     */
+    public function assertMailContainsAt($at, $contents, $message = null)
+    {
+        $this->assertThat($contents, new MailContainsConstraint($at), $message);
+    }
+
+    /**
+     * Asserts an email at a specific index contains the expected value within an Email getter
+     *
+     * @param int $at Email index
+     * @param int $expected Contents
+     * @param int $parameter Email getter parameter (e.g. "cc", "subject")
+     * @param string $message Message
+     * @return void
+     */
+    public function assertMailSentWithAt($at, $expected, $parameter, $message = null)
+    {
+        $this->assertThat($expected, new MailSentWithConstraint($at, $parameter), $message);
+    }
+
+    /**
+     * Asserts an email was sent to an address
+     *
+     * @param int $address Email address
+     * @param string $message Message
+     * @return void
+     */
+    public function assertMailSentTo($address, $message = null)
+    {
+        $this->assertThat($address, new MailSentToConstraint(), $message);
+    }
+
+    /**
+     * Asserts an email was sent from an address
+     *
+     * @param int $address Email address
+     * @param string $message Message
+     * @return void
+     */
+    public function assertMailSentFrom($address, $message = null)
+    {
+        $this->assertThat($address, new MailSentFromConstraint(), $message);
+    }
+
+    /**
+     * Asserts an email contains expected contents
+     *
+     * @param int $contents Contents
+     * @param string $message Message
+     * @return void
+     */
+    public function assertMailContains($contents, $message = null)
```

Comment:
Should there be assertions for html/text only messages? What about comparing a body to a known fixture?

Reply:
Are you thinking different methods for each type of message? Or, add a param here to check 'html', 'text', or 'both' (for either)?

I also like comparing to a fixture. There are honestly a lot of assertions we can add here, including checking attachments.

Reply:
Different methods would be easier from a documentation perspective and is more aligned with how PHPUnit works.